### PR TITLE
Add tests to pkg describe

### PR DESCRIPTION
**Pull request by janisz**

### CODE REVIEW

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

File: pkg/cmd/pkg/testdata/cosmos_chronos_describe_config.expected.json [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        },
+        "zk-path": {
+          "description": "Path in ZooKeeper for storing state (default = /state)",
+          "type": "string"
+        },
+        "zk-timeout": {
+          "default": 10000,
+          "description": "The timeout for ZooKeeper in milliseconds (default = 10000)",
+          "minimum": 0,
+          "type": "integer"
```
Comment (bamarni) [inline pkg/cmd/pkg/testdata/cosmos_chronos_describe_config.expected.json]: Is it really needed to have hundred of lines of configs for this unit test? It'd make it easier to understand if it was a simple package definition with only a few options, and wouldn't affect the test coverage / intent.

Comment (janisz) [reply pkg/cmd/pkg/testdata/cosmos_chronos_describe_config.expected.json]: Probably we don't need it but it was easiest way to start. Can you propose minimal valid package struct?

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

File: pkg/cmd/pkg/testdata/cosmos_chronos_describe_config.expected.json (hunk 1 above)
Comment (bamarni) [reply pkg/cmd/pkg/testdata/cosmos_chronos_describe_config.expected.json]: The helloworld test package we use in integration tests could be a minimal example (https://github.com/mesosphere/universe/blob/cli-test-v3-helloworld/repo/packages/H/helloworld/0/config.json).

File: pkg/cosmos/client.go [hunk 2]
```diff
@@ -18,13 +18,26 @@ import (
 	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
 )
 
-// Client is a diagnostics client for DC/OS.
-type Client struct {
+// Client is a Cosmos client abstraction
+type Client interface {
```
Comment (bamarni) [inline pkg/cosmos/client.go:22]: What is the reason for introducing this interface?

Comment (janisz) [reply pkg/cosmos/client.go:22]: This interface is used to inject mock inplementation in test.
@makkes requested this change https://github.com/dcos/dcos-core-cli/pull/360#discussion_r326988597

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

File: pkg/cmd/pkg/testdata/cosmos_chronos_describe_config.expected.json (hunk 1 above)
Comment (janisz) [reply pkg/cmd/pkg/testdata/cosmos_chronos_describe_config.expected.json]: Is it better now?

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3  Can you review, please?

**Review by bamarni — APPROVED** [review]

File: pkg/cmd/pkg/testdata/cosmos_chronos_describe_config.expected.json (hunk 1 above)
Comment (bamarni) [reply pkg/cmd/pkg/testdata/cosmos_chronos_describe_config.expected.json]: :+1:

File: pkg/cosmos/client.go (hunk 2 above)
Comment (bamarni) [reply pkg/cosmos/client.go:22]: I see, I had missed that as the file is hidden like other vendor files on Github.

File: pkg/cmd/pkg/testdata/describe.expected.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      "foo": "bar"
+    },
+    "description": "Package stub with only required fields",
+    "maintainer": "John Doe",
+    "marathon": {
+      "v2AppMustacheTemplate": "ewogICAgImlkIjogInt7aGVsbG93b3JsZC5pZH19IiwKICAgICJjbWQiOiAiZWNobyAnSGVsbG8gV29ybGQnIgp9"
+    },
+    "name": "helloworld",
+    "packagingVersion": "0.0.1",
+    "releaseVersion": 9223372036854775807,
```
Comment (bamarni) [inline pkg/cmd/pkg/testdata/describe.expected.json:14]: I hope it's not the real value we get when calling Cosmos :sweat_smile:

Comment (janisz) [reply pkg/cmd/pkg/testdata/describe.expected.json:14]: It's there to check for int32 overflow

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

File: pkg/cmd/pkg/package_describe_test.go [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/dcos/dcos-cli/pkg/mock"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_pkgDescribe(t *testing.T) {
```
Comment (makkes) [inline pkg/cmd/pkg/package_describe_test.go]: I personally don't like underscores in function names too much. Also IMHO test names should reflect what the test does.

Comment (janisz) [reply pkg/cmd/pkg/package_describe_test.go]: Fixed

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3  Can you review, please?

**Review by makkes — APPROVED** [review]

**Pull request merged**